### PR TITLE
Suggestion re introducing sc_group vs sc_dataset

### PR DESCRIPTION
**Pull request by falexwolf**
Extending discussion here: https://github.com/single-cell-data/matrix-api/issues/11

* This would change the sequence and first introduce `sc_group` & `sc_dataset` and then talk about the specification within `sc_group`.
* It would make identifications with existing formats on the level of `sc_group` and not on the level of the fields within `sc_group`.

There is also a question here regarding "allowing arbitrary collections of dataframes" vs. aligned multi-dimensional annotations (https://github.com/single-cell-data/matrix-api/issues/3).

### CODE REVIEW

File: specification.md
Comment (ambrosejcarr) [inline specification.md:50]:
```suggestion
* Any other number of auxiliary **dataframes** and **arrays**.
```

We do need obsm and varm. I suggest we address that in a separate PR, which would unblock merging of this one.

I've update #3 with some thoughts and questions for you.

**Review by ambrosejcarr — APPROVED** [review]
These changes look good and I'd be happy to see them merged.

@aaronwolen Could you provide a second opinion? Is there anything here you think should be changed?

**Comment by aaronwolen** [comment]
Looks great. I agree that starting with descriptions of the higher-level components and including their anndata/seurat/bioc analogs will help readers get oriented more quickly. 

Related but probably for a separate PR:
- I think we should bump-up the `matrix.png` diagram to come before *Notes:*
- and some of those bullets could be folded into the revised descriptions (e.g., *There can be more than one `X` arrays...*)

**Comment by ambrosejcarr** [comment]
Thanks for this suggestion @falexwolf !

**Comment by falexwolf** [comment]
Thanks for merging! 😄

**Pull request merged**
